ios/OpenBot/OpenBot/Controllers/HomePage/HomePageViewController.swift

```diff
@@ -14,35 +14,70 @@ var viewControllerName: String?
 let gameController = GameController.shared
 var leadingConstraint = NSLayoutConstraint()
 var isClientConnected: Bool = false
+let bottomSheet = UIView();
+let whiteSheet = UIView(frame: UIScreen.main.bounds)
+
+class HeaderView: UICollectionReusableView {
+    var label: UILabel!
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        label = UILabel(frame: CGRect(x: 0, y: 0, width: frame.size.width, height: frame.size.height+30))
+        label.textAlignment = .left
+        label.font = UIFont.boldSystemFont(ofSize: 20) // Change the font to bold and size to 20
+        self.addSubview(label)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
 
-class HomePageViewController: CameraController {
+class HomePageViewController: CameraController,UICollectionViewDataSource,UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
     @IBOutlet weak var bluetooth: UIButton!
     @IBOutlet weak var settings: UIButton!
     @IBOutlet weak var titleLabel: UILabel!
     @IBOutlet var modesCollectionView: UICollectionView!;
+    let sectionHeaders = ["  General", "  AI"]
 
     /// Called after the view controller has loaded.
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        UITabBar.appearance().tintColor = traitCollection.userInterfaceStyle == .dark ? UIColor.white: UIColor.black;
         bluetoothDataController.shared.startScan()
         DeviceCurrentOrientation.shared.findDeviceOrientation()
         setUpTitle();
         let layout = UICollectionViewFlowLayout();
         layout.collectionView?.layer.shadowColor = Colors.gridShadowColor?.cgColor
         layout.collectionView?.layer.shadowOpacity = 1
         if currentOrientation == .portrait {
-            layout.itemSize = resized(size: CGSize(width: width * 0.42, height: width * 0.42), basedOn: dimension)
+            layout.itemSize = resized(size: CGSize(width: width * 0.15, height: width * 0.15), basedOn: dimension)
         } else {
-            layout.itemSize = resized(size: CGSize(width: width * 0.42, height: width * 0.42), basedOn: dimension)
+            layout.itemSize = resized(size: CGSize(width: width * 0.15, height: width * 0.15), basedOn: dimension)
         }
-
+        layout.collectionView?.backgroundColor = Colors.voltageDividerColor
```

Comment:
Why is this color called voltageDividerColor?

Reply:
It is the green gauge in the freeroam fragment.

Reply:
Seems better practice to define a separate color (even if same RGB value) or give it a neutral name.